protocol/testing/e2e/gov/add_new_market_test.go

```diff
@@ -0,0 +1,256 @@
+package gov_test
+
+import (
+	"testing"
+
+	"github.com/cometbft/cometbft/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	clobtest "github.com/dydxprotocol/v4-chain/protocol/testutil/clob"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/encoding"
+	perptest "github.com/dydxprotocol/v4-chain/protocol/testutil/perpetuals"
+	pricestest "github.com/dydxprotocol/v4-chain/protocol/testutil/prices"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	delaymsgtypes "github.com/dydxprotocol/v4-chain/protocol/x/delaymsg/types"
+	perptypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	pricestypes "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddNewMarketProposal(t *testing.T) {
+	testMarketParam1001 := pricestest.GenerateMarketParamPrice(
+		pricestest.WithId(1001),
+	)
+	testClobPair1001 := clobtest.GenerateClobPair(
+		clobtest.WithId(1001),
+		clobtest.WithPerpetualId(1001),
+		clobtest.WithStatus(clobtypes.ClobPair_STATUS_INITIALIZING),
+	)
+	testPerpetual1001 := perptest.GeneratePerpetual(
+		perptest.WithId(1001),
+		perptest.WithMarketId(1001),
+	)
+	msgUpdateClobPair1001ToActive := &clobtypes.MsgUpdateClobPair{
+		Authority: authtypes.NewModuleAddress(delaymsgtypes.ModuleName).String(),
+		ClobPair: *clobtest.GenerateClobPair(
+			clobtest.WithId(1001),
+			clobtest.WithPerpetualId(1001),
+			clobtest.WithStatus(clobtypes.ClobPair_STATUS_ACTIVE),
+		),
+	}
+
+	tests := map[string]struct {
+		proposedMsgs             []sdk.Msg
+		updateClobDelayBlocks    uint32
+		expectSubmitProposalFail bool
+		expectedProposalStatus   govtypesv1.ProposalStatus
+	}{
+		"Success with 4 standard messages": {
+			proposedMsgs: []sdk.Msg{
+				&pricestypes.MsgCreateOracleMarket{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params:    testMarketParam1001.Param,
+				},
+				&perptypes.MsgCreatePerpetual{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params:    testPerpetual1001.Params,
+				},
+				&clobtypes.MsgCreateClobPair{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					ClobPair:  *testClobPair1001,
+				},
+				&delaymsgtypes.MsgDelayMessage{
+					Authority:   authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Msg:         encoding.EncodeMessageToAny(t, msgUpdateClobPair1001ToActive),
+					DelayBlocks: 10,
```

Comment:
can we add a test case with different DelayBlocks value? specifically

* DelayBlocks: 0
* DelayBlocks: 1